src/main/java/edu/wpi/grip/core/events/UnexpectedThrowableEvent.java

```diff
@@ -0,0 +1,37 @@
+package edu.wpi.grip.core.events;
+
+/**
+ * Event should be thrown when Unexpected Throwable ends up in an {@link Thread#uncaughtExceptionHandler}.
+ */
+public final class UnexpectedThrowableEvent {
+    private final Throwable throwable;
+    private final boolean fatal;
+
+    /**
+     * @param throwable The throwable that was caught.
+     * @param fatal     True if this cause the application to quit forcibly.
+     *                  If the throwable is an {@link Error} than this is automatically true. Defaults to false.
+     */
+    public UnexpectedThrowableEvent(Throwable throwable, boolean fatal) {
```

Comment:
I plan to. In a different PR. I'm trying to go for small pull requests that are easy to review and merge.